index.js

```diff
@@ -253,18 +284,21 @@ Object.defineProperty(File.prototype, 'extname', {
 
 Object.defineProperty(File.prototype, 'path', {
   get: function() {
-    return this.history[this.history.length - 1];
+    var last = this.history[this.history.length - 1];
+    return this.isDirectory() ? endWithSep(last) : last;
   },
-  set: function(path) {
-    if (typeof path !== 'string') {
-      throw new Error('path should be string');
+  set: function(newPath) {
+    if (typeof newPath !== 'string') {
```

Comment:
Was asking this myself, and came to conclusion that file in the process of being created that has content, but doesn't have a name yet is definitely a use case.

Example: A GUI app where user creates a new file, starts writing the content, but doesn't name it yet.

Of course, this could be also handled by assigning the file a temporary UID until it's named, but the developer could use the falsy path as a check that file is not named, and now he is forced to add a custom flag for that, and keep it with sync. He'll also have to have a special handling for deletion of the filename (everyone deletes the previous name when renaming) so it won't throw errors. It'd be quite messy.

So with this is mind, I lean with allowing empty `path`.